### ISSUE TITLE
Improvements

### ISSUE THREAD

**Issue by marckaraujo** [Kairos-Project/kairos-framework#4]
Hello! Thanks for your framework! its awesome!
I was really looking hard to find something like fragments for JavaFX and it really dont have anything!

Ok, I´d like to sugest some improvements, I think that framework is really useful for Java Desktop developers but they dont know it, probably because never used to develop for Android and how good it is. Job done by Google and not Oracle.

So we need to bring some Java Desktop developers for our side. 

- Maven support (They still use maven only)
- Examples (Improve the examples)
- Visual Documentation (Setup a gif to show how cool is the framework, ex: https://github.com/Ramotion/folding-cell)
- Fragments - We need to show how good this fragments is, and JavaFX dont have nothing to solve this at moment.   
- Gradle - https://github.com/FibreFoX/javafx-gradle-plugin, the actual gradle plugin is outdated and non-maintained

I will try to make some PR, about this, let me know what you think?

Thanks

**Comment by abdlquadri** [comment]
Hi @marckaraujo , I think this project is dead as it has not received any update in the last year. All efforts to reach the project creator also proved abortive.

I  recommend you make your improvement in your own fork. Please let us know where and when that is so others can contribute.

Thank You.

**Comment by marckaraujo** [comment]
@abdlquadri You are right, I have made a contact with the project owner and he just avoid me. I am still working with this library, I did some improvements. If you liked it, we can move this foward in a new fork.

**Comment by abdlquadri** [comment]
Yes please. I just wish @marconi1992 well because he has done a great Job. I hope he is fine.

unfortunately he didn't release this before going quite. https://www.youtube.com/watch?v=-XHOp6plI3g

**Comment by marconi1992** [comment]
@marckaraujo @abdlquadri I'm fine, you can fork and improve this project, currently I'm busy with other priorities therefore I can't monitor it.

**Comment by abdlquadri** [comment]
Wao! Thanks! for taking time to respond. 

Pls can you just push to github  your latest work on the project. Particularly meaningful motion. 

I promise not to worry you with support for it.  Thanks. Wish you the best.

**Comment by abdlquadri** [comment]
Hi @marconi1992 , I wish you saw my comment above. Thanks!

**Comment by marckaraujo** [comment]
@marconi1992 , can you give to us access to the website? 

The link to new fork: https://github.com/marckaraujo/kairos-framework
I will create a special thanks to you, if you create a message here too, just to send the old users to the new fork.

Really thanks

**Comment by marckaraujo** [comment]
@abdlquadri I also need to help with this haha, dont run XD

**Comment by abdlquadri** [comment]
@marckaraujo, @marconi1992 just merged your PR. Yeah I am here.

BTW you heard about https://github.com/jfoenixadmin/JFoenix ? Better we monitor them so we don't duplicate what they have done.

I think the focus of this project should be on the Framework. We can reuse UI components from _JFoenix_

@marconi1992 I still await the https://www.youtube.com/watch?v=-XHOp6plI3g Meaningful motion code. Thanks!

**Comment by marckaraujo** [comment]
@abdlquadri, I agree. We should focus on framework core, it should be a framework that you can code just like android and use android´s libraries.

**Comment by marconi1992** [comment]
@abdlquadri  the meaningful motion consist two animations the first before to the fragment transaction and the other after that, really is not a part of the framework ,like in android not is a simple animation in the transition between fragments. I'll add an example in this repo  https://github.com/Kairos-Project/kairos-example

**Comment by abdlquadri** [comment]
@marconi1992 Thanks! Will expect.

**Comment by marconi1992** [comment]
@abdlquadri  The example is in this release https://github.com/Kairos-Project/kairos-example/releases/tag/1.0.4

The related code is here https://github.com/Kairos-Project/kairos-example/blob/master/src/main/java/fragments/AppFragment.java.

In the example the first fragment before to start the transaction sends the position and size of the card component, when the second fragment is attached set these values to the second card and starts the animation. https://github.com/Kairos-Project/kairos-example/blob/master/src/main/java/fragments/DetailsFragment.java

**Comment by abdlquadri** [comment]
@marconi1992 Thanks.  Really appreciated. I will  run it when I get to my PC.  Thanks.

**Comment by marckaraujo** [comment]
@abdlquadri, do you have access now to this repository?

I was working to improve this library. And I really want to think what is our main propose.

Case 1: Core framework like Java for Android
Because if we are trying to create library as Google did with Java for Android. It is really a big job.
And I dont know if even we do it, we can get enough value to people work with our framework.
Thats because, a lot of other libraries should be done using our framework as core.

Case 2: Alter Java library for Desktop
I dont know how the license of Java SDK for Android works, but a lot of companies have their own version. So if we alter the Java SDK for Android and bring that to Javafx in the way that any Java framework developed for Android could be used. Like: [Picasso](https://github.com/square/picasso)

Why I user this library atm? Fragments, this feature in Android is **Amazing** and with Kairos we can now use too.

*First thing to do also with this library is merge kairos-framework with kairos-core.

**Comment by abdlquadri** [comment]
Hi @marckaraujo , I think the main idea of this project is to allow for JavaFx UI development like that of android.

Basically the selling points are `Activity`, `Fragment`, `Intent` etc.

The other alternatives in JavaFx are too complicated and unfamiliar IMHO, i.e _DataFX_, _afterburner fx_

Also since @marconi1992 seems responsive now, do we still go ahead and fork?

Also @marconi1992 can chime in on the future direction and features.

I am currently working on a meaningful motion demo using JFoenix and Kairos

**Comment by marconi1992** [comment]
@marckaraujo  I also think we should move kairos core to kairos framework , maybe add an IoC like guice for dependency injection of services into the Activities and Fragments  and add some unit tests.

@abdlquadri @marckaraujo  We could work together, I'll  send you an invitation to the organization and during the week we could define the features for a minimun viable product.

The JFoenix framework can save us time to develop material design components and just focus in develop components coupled to the application lifecycle like DrawerLayout, Toolbar, ViewPager and Tabs.
